src/main/java/com/snowflake/kafka/connector/internal/streaming/DefaultStreamingConfigValidator.java

```diff
@@ -0,0 +1,222 @@
+package com.snowflake.kafka.connector.internal.streaming;
+
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.*;
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.INGESTION_METHOD_OPT;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig;
+import com.snowflake.kafka.connector.Utils;
+import com.snowflake.kafka.connector.internal.BufferThreshold;
+import com.snowflake.kafka.connector.internal.KCLogger;
+import com.snowflake.kafka.connector.internal.parameters.InternalBufferParameters;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.apache.kafka.common.config.ConfigException;
+
+public class DefaultStreamingConfigValidator implements StreamingConfigValidator {
```

Comment:
This class contains everything from the `StreamingUtils.java` validation. The only difference is adding `SingleBufferConfigValidator` that logs warnings and `DoubleBufferConfigValidator` that performs validations as before.

It would be great to unit test this class separately from `ConnectorConfigValidatorTest` but I won't do it in this PR.